### ISSUE TITLE
Empty directories are not removed from cache after deletion

### ISSUE THREAD

**Issue by vikonnikov** [dgilland/hashfs#11]
Function `mkpath()` from `distutils.dir_util` saves all paths in global variable `_path_created`, but `HashFS` doesn't remove paths of removed directories from this variable.

```
import io
import os
from hashfs import HashFS

# from distutils.dir_util import _path_created

root = './hashfs'
if not os.path.exists(root):
    os.mkdir(root)

fs = HashFS(root, algorithm='sha256', depth=4)

for i in range(10):
    f = io.BytesIO(os.urandom(1024))

    address = fs.put(f)
    fs.delete(address.id)

    # print _path_created
```

```
Traceback (most recent call last):
  File "hashfsbug.py", line 15, in <module>
    address = fs.put(f)
  File ".../env/local/lib/python2.7/site-packages/hashfs/hashfs.py", line 67, in put
    filepath, is_duplicate = self._copy(stream, id, extension)
  File ".../env/local/lib/python2.7/site-packages/hashfs/hashfs.py", line 85, in _copy
    self.makepath(os.path.dirname(filepath))
  File ".../env/local/lib/python2.7/site-packages/hashfs/hashfs.py", line 232, in makepath
    mkpath(path, mode=self.dmode)
  File "/usr/lib/python2.7/distutils/dir_util.py", line 76, in mkpath
    "could not create '%s': %s" % (head, exc.args[-1]))
distutils.errors.DistutilsFileError: could not create '...hashfs/d/7': No such file or directory
```

Hotfix without cache:

```
import os
import hashfs

class HashFS(hashfs.HashFS):
    def makepath(self, path):
        if not os.path.exists(path):
            os.makedirs(path, self.dmode)
```

**Comment by dgilland** [comment]
This is fixed in 1f3d5ba714fb88f6f372a1ab2ba49ee97b952423 but hasn't been released. I'll work on releasing that this week.

**Comment by dgilland** [comment]
I've released `v0.7.1` which includes the fix.
